src/sfnewton.h

```diff
@@ -61,7 +61,7 @@ class SFNewton {
 	Real minimum;
 
 	virtual void residuals(Real*,Real*); //x,g
-	virtual void inneriteration(Real*,Real*,float*, Real, Real&, Real, int); //x g accuracy nvar
+	virtual void inneriteration(Real*,Real*,Real*, Real, Real&, Real, int); //x g accuracy nvar
```

Comment:
Looking at it again I'm pretty sure this was also a fix for CUDA related issues.

I did test it, but it may not be a bad idea to test it against the current master branch to be 100% sure. I'll make sure to do so before merging to master. This is used in pseudo-hessian, right?

Reply:
Yes, it is used in definitely used in pseudohessian, hessian and may be diis. There is also svd (single value decomposition) which might not accept user defined types.

Reply:
Gotcha, I will look into it and keep you posted.

Reply:
My testing worked fine, except for an unrelated issue with Hessian that was introduced with the constraintfields.

- When using Pseudohessian the profiles only differ after the 11th digit, even at the steepest gradient - which would be expected when using higher precision.
- Hessian gives an unrelated sysmalloc issue (call site solve_scf.cpp:851)
- DIIS gives 100% the same result

Reply:
The Hessian issue is caused by an un-malloced g variable. Newton uses a miriad of scroped and global g variables, that may or may not be malloced at a certain point. I came up with a fix but I'm still finding out if it also gives the correct result.